server/test/com/cloud/user/DomainManagerImplTest.java

```diff
@@ -134,4 +164,67 @@ public void testFindDomainByIdOrPathValidId() {
+        domainManager.deleteDomain(DOMAIN_ID, testDomainCleanup);
+    }
+
+    @Test(expected=PermissionDeniedException.class)
+    public void testDeleteDomainRootDomain() {
+        Mockito.when(_domainDao.findById(Domain.ROOT_DOMAIN)).thenReturn(domain);
+        domainManager.deleteDomain(Domain.ROOT_DOMAIN, testDomainCleanup);
+    }
+
+    //TODO testDeleteDomainCleanup
+
+    @Test
+    public void testDeleteDomainNoCleanup() {
+        domainManager.deleteDomain(DOMAIN_ID, testDomainCleanup);
+        Mockito.verify(domainManager).deleteDomain(domain, testDomainCleanup);
+        Mockito.verify(domainManager).checkDomainAccountsNetworksAndResourcesBeforeRemoving(domain);
+        Mockito.verify(domainManager).cleanupDomainOfferings(DOMAIN_ID);
+        Mockito.verify(lock).unlock();
+        Assert.assertEquals(false, domainManager.getRollBackState());
+    }
+
+    @Test
+    public void testCheckDomainAccountsNetworksAndResourcesBeforeRemovingRemoveDomain() {
+        domainManager.checkDomainAccountsNetworksAndResourcesBeforeRemoving(domain);
+        Mockito.verify(domainManager).publishRemoveEventsAndRemoveDomain(domain);
+    }
+
+    @Test(expected=CloudRuntimeException.class)
+    public void testCheckDomainAccountsNetworksAndResourcesBeforeRemovingDontRemoveDomain() {
+        domainNetworkIds.add(2l);
+        domainManager.checkDomainAccountsNetworksAndResourcesBeforeRemoving(domain);
+        Mockito.verify(domainManager).failRemoveOperation(domain, domainAccountsForCleanup, domainNetworkIds, false);
+    }
+
+    @Test
+    public void testPublishRemoveEventsAndRemoveDomainSuccessfulDelete() {
+        domainManager.publishRemoveEventsAndRemoveDomain(domain);
+        Mockito.verify(_messageBus).publish(Mockito.anyString(), Matchers.eq(DomainManager.MESSAGE_PRE_REMOVE_DOMAIN_EVENT),
+                Matchers.eq(PublishScope.LOCAL), Matchers.eq(domain));
+        Mockito.verify(_messageBus).publish(Mockito.anyString(), Matchers.eq(DomainManager.MESSAGE_REMOVE_DOMAIN_EVENT),
+                Matchers.eq(PublishScope.LOCAL), Matchers.eq(domain));
+        Mockito.verify(_domainDao).remove(DOMAIN_ID);
+    }
+
+    @Test(expected=CloudRuntimeException.class)
+    public void testPublishRemoveEventsAndRemoveDomainExceptionDelete() {
+        Mockito.when(_domainDao.remove(DOMAIN_ID)).thenReturn(false);
+        domainManager.publishRemoveEventsAndRemoveDomain(domain);
+        Mockito.verify(_messageBus).publish(Mockito.anyString(), Matchers.eq(DomainManager.MESSAGE_PRE_REMOVE_DOMAIN_EVENT),
+                Matchers.eq(PublishScope.LOCAL), Matchers.eq(domain));
+        Mockito.verify(_messageBus, Mockito.never()).publish(Mockito.anyString(), Matchers.eq(DomainManager.MESSAGE_REMOVE_DOMAIN_EVENT),
+                Matchers.eq(PublishScope.LOCAL), Matchers.eq(domain));
+        Mockito.verify(_domainDao).remove(DOMAIN_ID);
+    }
+
+    @Test(expected=CloudRuntimeException.class)
+    public void testFailRemoveOperation() {
```

Comment:
Thanks! I removed `expected` from `@Test` annotation and added a catch block to assert exception class. It is a nasty method to test indeed :)